### PR TITLE
Organize changelog

### PR DESCRIPTION
**Pull request by dahlia**
I organized changelogs into four categories:

- Backward-incompatible interface changes<sup>※</sup>
- Added interfaces
- Behavioral changes<sup>※</sup>
- Bug fixes

----

※ This should bump the major version (or the minor version until 1.0.0).

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -6,79 +6,93 @@ Version 0.3.0
...
-    of `IndexOutOfRangeException`.  [[#210]]
+ -  Removed `KeyEquals()` methods from all classes and structs.  [[#216]]
+ -  `Swarm` class now does not implement `IEquatable<Swarm>` anymore and
+    its `Equals(object)` method and `GetHashCode()` method became to have
+    default behavior of `object` class.  [[#216]]
+ -  The signature of `IStore.PutBlock<T>(Block<T>)` method was changed to
+    `PutBlock<T>(Block<T>, Address)`.  [[#189], [#197]]
+ -  The type of `Block<T>.Difficulty` is changed to `long` instead of `int`, and
+    related classes method parameters and field types have changed accordingly.
+ -  Removed `HashDigest.HasLeadingZeroBits()` method.  [[#213]]
```
Comment (earlbread) [inline CHANGES.md]: I think
```
 -  `BlockPolicy<T>` constructor became to receive the `minimumDifficulty`	
     and the mining `difficultyBoundDivisor`.  [[#213]]
```
is missing.

**Comment by dahlia** [comment]
@earlbread @longfin I added a missing changelog.  Please review this again!

**Comment by earlbread** [comment]
The changelog below does not seem to exist yet.
```
 -  `BlockPolicy<T>` constructor became to receive the `minimumDifficulty`	
     and the mining `difficultyBoundDivisor`.  [[#213]]
```

**Comment by dahlia** [comment]
@earlbread @longfin Sorry, fixed again. Please review this again!

**Pull request merged**